### ISSUE TITLE
permissions of my own grain changed after renaming document

### ISSUE THREAD

**Issue by jadeqwang** [rchrd2/TextEditor#2]
https://youtu.be/-wuoEpa5zj0

Steps to repro:
1. Created a doc and type a bunch of stuff
2. Went to lunch for a while
3. Came back, copied text into a fresh instance of TextEditor
4. Renamed first TextEditor grain
5. Attempt to type to find that it's suddenly read-only.

**Comment by jadeqwang** [comment]
This is fixed by closing the grain and re-opening it, in case that helps.

**Comment by rchrd2** [comment]
Hi Jade. Thanks for pointing this out. I haven't run into this myself, but I'll take a look at it later. Great to see someone else using it!

**Comment by jadeqwang** [comment]
Yeah, I actually rely on TextEditor over Etherpad for my own work these days because 1) way faster startup time, and 2) I don't really use bold/italics/comments that much anyway since I mostly format my own writing in Markdown.

**Comment by ocdtrekkie** [comment]
I haven't migrated entirely over to it, and for some stuff, Etherpad makes sense. But I love not having excess features (and UI) I don't need. So for quick notes for myself and stuff, the fast loading absolutely wins out. Really happy you made this.

**Comment by rchrd2** [comment]
Great. I'd love to hear any feature ideas you may have, while keeping it simple. Here are some of my ideas:
- [ ] Use a op-transform library (sharejs) so two people can edit at the same time without glitches
- [ ] Maybe add a subtle menu to add document settings (font styles/size/etc)
- [ ] Show who else is looking at the document
- [ ] indicate when internet connection is lost (and hopefully still allow editing in offline mode)

**Comment by ocdtrekkie** [comment]
+1. sharejs in particular is always ideal, because I think many Sandstorm users will assume/expect apps to 'just work' when they are shared with others.

My big advice is that I think the minimal UI is a key perk (I haven't tried it yet, but my imagination suspects your app may be nicer on mobile users than Etherpad), and of course the speed. So a very minimal UI for the above might be in a right corner like "O V Menu", where O is an icon indicating you're connected, V is an icon indicating the number of other people opening the document (clicking it would display who of course), and a self-explanatory menu button for display options (font size is good for accessibility at bare minimum) and maybe a download button if people want to download a 'txt' format file.

If you wanna do a much more capable text editor beyond that, Etherpad could probably use some good competition on the Sandstorm market, but you should either way retain an incredibly simple text client as well because it's cool/handy.

**Comment by rchrd2** [comment]
@ocdtrekkie Thanks for the suggestions. I'm onboard with keeping it nice and simple.

@jadeqwang Coincidentally, this same issue occurred to me yesterday while I was taking some notes on my phone with this app. I think it has to do with losing internet connection and maybe the socket connection. I need to investigate further, but good news is I can reproduce it.

**Comment by jadeqwang** [comment]
hey @rchrd2 ,

Yep, I ran into it again today with a different TextEditor document. However, there was no renaming step this time. Just had it open for a long time, went to a bunch of meetings, came back and couldn't continue working on a blog post until I closed it and reopened it.

**Comment by rchrd2** [comment]
@jadeqwang Sorry about that. I have experienced this a few times too. I don't know what it is exactly, yet. But I think it might have to do with `meteor-headers`, which I am using to check for permissions. I'm wondering if those headers get lost sometimes after a reconnect. 

I'm finishing up a new version of the editor that has a settings drawer and allows styling of the document. After this is completed (almost done), I want to fix this bug. And after that, comes sharejs, so two people can easily edit at the same time.
